### PR TITLE
Expose `parseColumnType` function

### PR DESCRIPTION
**Pull request by slvrtrn**
## Summary

Extracted and exported the `parseColumnType` method from the #257 branch. Related to #273.
This is useful not only for RowBinary/Native protocols but also for TS interface codegens, so it is probably worth adding it right away.

For example:
- Input type: `Array(Array(DateTime('Etc/GMT-5')))`
- Output object:
```ts
{
  sourceType: `Array(Array(DateTime('Etc/GMT-5')))`,
  type: 'Array',
  dimensions: 2,
  value: {
    type: 'DateTime',
    timezone: 'Etc/GMT-5',
    sourceType: `DateTime('Etc/GMT-5')`,
  },
}
```

Currently supported types to parse:

* `Bool`
* `UInt8`
* `Int8`
* `UInt16`
* `Int16`
* `UInt32`
* `Int32`
* `UInt64`
* `Int64`
* `UInt128`
* `Int128`
* `UInt256`
* `Int256`
* `Float32`
* `Float64`
* `String`
* `FixedString(N)`
* `UUID`
* `Date`
* `Date32`
* `IPv4`
* `IPv6`
* `Array` (arbitrary nesting)
* `Map` (arbitrary nesting)
* `Tuple` (arbitrary nesting)
* `Enum` (including enums in arrays, maps, tuples)
* `DateTime` (including timezones)
* `DateTime64` (including timezones and various precisions)
* `Decimal`
* `Nullable`, `LowCardinality` combinations

Currently, not supported (will throw):

* `Geo`
* `(Simple)AggregateFunction`
* `Nested`
* Old/new `JSON`
* `Dynamic`
* `Variant`

Please take a look at the tests for more examples.
The tests are not fully complete, so this will be somewhat "experimental" still.

## Checklist
- [ ] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

File: packages/client-node/src/index.ts [hunk 1]
```diff
@@ -46,4 +46,17 @@ export {
...
+  type ParsedColumnFixedString,
+  type ParsedColumnNullable,
+  type ParsedColumnDecimal,
+  type ParsedColumnDateTime,
+  type ParsedColumnDateTime64,
+  type ParsedColumnArray,
+  type ParsedColumnTuple,
+  type ParsedColumnMap,
+  type ParsedColumnType,
+  parseColumnType,
```
Comment (mshustov) [inline packages/client-node/src/index.ts:61]: Neither nodejs nor web clients use any of these by default. Should we consider releasing it as a separate package? 
What benefits does the current distribution model have? I see a few disadvantages: larger distro size and harder to explore the original source code

File: packages/client-common/__tests__/unit/parse_column_types_enum.test.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    })
+  })
+
+  it('should throw when Enum has no values to parse', async () => {
+    // The minimal allowed Enum definition is Enum8('' = 0), i.e. 6 chars inside.
+    const allEnumTypeArgs: string[][] = enumTypes.map(([enumType]) => [
+      `${enumType}()`,
+      `${enumType}(')`,
+      `${enumType}('')`,
+      `${enumType}('' )`,
```
Comment (mshustov) [inline packages/client-common/__tests__/unit/parse_column_types_enum.test.ts:77]: ~could you add a test that an empty string is a valid value?~ found it below

**Review by mshustov — APPROVED** [review]

File: packages/client-node/src/index.ts (hunk 1 above)
Comment (slvrtrn) [reply packages/client-node/src/index.ts:61]: > Neither nodejs nor web clients use any of these by default. Should we consider releasing it as a separate package?

If we are ever to add Native or RowBinaryWithNamesAndTypes, then this parser will be required anyway. Or do you mean that it should not be exported from the Node.js client, only in common?

> harder to explore the original source code

WDYM?
